staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -414,7 +414,8 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 		err           error
 	)
 
-	if node.Spec.ProviderID == "" {
+	_, instancesV2Enabled := cnc.cloud.InstancesV2()
+	if node.Spec.ProviderID == "" && !instancesV2Enabled {
```

Comment:
This probably needs a comment for why we're skipping the provider ID check when instance v2 is enabled.

Reply:
@andrewsykim gotcha! 
updated the notes